### ISSUE TITLE
Nginx in docker on zfs must now use 'senfile off;'

### ISSUE THREAD

**Issue by jfburdet** [openzfs/zfs#13013]
I folks. I have an ArchLinux with zfs running a dokuwiki using this docker image : ghcr.io/linuxserver/dokuwiki:version-2020-07-29
It worked very well since a long time.

But a few days ago, it stopped working  and started filling nginx log (in docker) with this kind of message : 

```
[alert] 386#386: *310 sendfile() failed (22: Invalid argument) while sending response to client,
```

when serving static files.

After a long time searching, I simply changed an option in nginx (in docker) : 'senfile off;' and the problem was gone.

I believe this could be linked to something bad happening between my Linux version and zfs. I believe it brokes when upgrading from 2.1.1 to 2.1.2

'hope this report can be usefull.

```
[root@orion ~]# zfs version
zfs-2.1.2-1
zfs-kmod-2.1.2-1
[root@orion ~]# uname -a
Linux orion 5.16.2-arch1-1 #1 SMP PREEMPT Thu, 20 Jan 2022 16:18:29 +0000 x86_64 GNU/Linux
```

**Comment by rincebrain** [comment]
#12971 

No current release claims 5.16 support, and as it turns out, it's broken.

**Comment by jfburdet** [comment]
Hmm, this look like make a duplicate issue :-( 
Sorry for that

**Comment by jfburdet** [comment]
Leaving it here for info : 

Seems to be fixed with : 

```
$ zfs version
zfs-2.1.2-1
zfs-kmod-2.1.2-1
$ uname -a
Linux orion 5.16.5-arch1-1 #1 SMP PREEMPT Tue, 01 Feb 2022 21:42:50 +0000 x86_64 GNU/Linux
```

**Comment by rincebrain** [comment]
Stock 2.1.2,definitely does not have the fixes for 5.16, and the archzfs packages don't immediately appear to have cherrypicked it, so...I'm not sure why it would work for you now?

**Comment by jfburdet** [comment]
Hhmm, I made a try but was tricking in believing it was fixed due to my browser cache . 

So no, you're right, bug is still here, and i'm still using "sendfile off" in my nginx config.
